canonicalwebteam/discourse/parsers/engage.py

```diff
@@ -155,3 +157,103 @@ def _parse_related(self, tags):
         """
         index_list = [item for item in self.metadata if item["tags"] in tags]
         return index_list
+
+    def _parse_engage_metadata(self, index_soup, section_name):
+        """
+        Given the HTML soup of an index topic
+        extract the metadata from the name designated
+        by section_name
+
+        This section_name section should contain a table
+        (extra markup around this table doesn't matter)
+        e.g.:
+
+        <h1>Metadata</h1>
+        <details>
+            <summary>Mapping table</summary>
+            <table>
+            <tr><th>Column 1</th><th>Column 2</th></tr>
+            <tr>
+                <td>data 1</td>
+                <td>data 2</td>
+            </tr>
+            <tr>
+                <td>data 3</td>
+                <td>data 4</td>
+            </tr>
+            </table>
+        </details>
+
+        This will typically be generated in Discourse from Markdown similar to
+        the following:
+
+        # Redirects
+
+        [details=Mapping table]
+        | Column 1| Column 2|
+        | -- | -- |
+        | data 1 | data 2 |
+        | data 3 | data 4 |
+
+        The function will return a list of dictionaries of this format:
+        [
+            {"column-1": "data 1", "column-2": "data 2"},
+            {"column-1": "data 3", "column-2": "data 4"},
+        ]
+        """
+        metadata_soup = self._get_section(index_soup, section_name)
+
+        topics_metadata = []
+        metadata_errors = []
+        if metadata_soup:
```

Comment:
This is a long and confusing block of BeautifulSoup logic (as they all are). Could you please add a comment above here to explain the markup structure you're parsing and what you're doing with it?